haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs

```diff
@@ -211,6 +216,113 @@ encodeSequence encodeItem s =
     encodeListLen (fromIntegral $ Seq.length s)
         <> foldMap encodeItem s
 
+convertSha256Hash :: CBOR.Term -> Maybe SHA256.Hash
+convertSha256Hash (CBOR.TBytes bs)
+    | BS.length bs == SHA256.digestSize = Just $ SHA256.Hash (FBS.fromByteString bs)
+    | otherwise = Nothing
+convertSha256Hash _ = Nothing
+
+encodeSha256Hash :: SHA256.Hash -> Encoding
+encodeSha256Hash (SHA256.Hash h) =
+    encodeBytes (FBS.toByteString h)
+
+-- | Convert CBOR.Term to a hex-encoded string
+cborTermToHex :: CBOR.Term -> Text
+cborTermToHex term =
+    let bs = CBOR.toStrictByteString $ CBOR.encodeTerm term
+    in  TextEncoding.decodeUtf8 (Base16.encode bs)
+
+-- | Convert a hex-encoded string to a CBOR.Term.
+hexToCborTerm :: Text -> Either String CBOR.Term
+hexToCborTerm hexText = do
+    bs <- Base16.decode (TextEncoding.encodeUtf8 hexText)
+    decodeTerm bs
+  where
+    decodeTerm bs =
+        case CBOR.deserialiseFromBytes CBOR.decodeTerm (LBS.fromStrict bs) of
+            Left err -> Left $ "Failed to decode CBOR term: " ++ show err
+            Right ("", term) -> Right term
+            Right (remaining, _) -> Left $ "Extra bytes after decoding CBOR term: " ++ show (LBS.length remaining)
```

Comment:
Could move these helper functions to local scopes instead. I left them here as they might be generally applicable in future composite types, but if we don't want that I can move them.